pkg/resources/opnicluster/services.go

```diff
@@ -578,7 +577,7 @@ func (r *Reconciler) gpuCtrlDeployment() (runtime.Object, reconciler.DesiredStat
 		Name:      "data",
 		MountPath: "/var/opni-data",
 	}
-	deployment.Spec.Template.Spec.RuntimeClassName = pointer.String("nvidia")
+	deployment.Spec.Template.Spec.RuntimeClassName = r.opniCluster.Spec.Services.GPUController.RuntimeClass
```

Comment:
Default logic should change if the GPUOperator feature gate is enabled such that:
- If the feature is enabled and RuntimeClass==nil, RuntimeClassName = "nvidia"
- If the feature is disabled and RuntimeClass==nil, RuntimeClassName = ""